cmd/flags/auth.go

```diff
@@ -1,10 +1,183 @@
+	authOICDClaimUsernameEnv      = "AUTH_OICD_CLAIM_USERNAME"
+)
+
+const (
+	authOICDClaimGroupFlag     = "auth-oidc-claim-groups"
+	authOICDClaimGroupViperKey = "auth.oidc.claim.groups"
+	authOICDClaimGroupDefault  = "qkm.auth.groups"
+	authOICDClaimGroupEnv      = "AUTH_OICD_CLAIM_GROUPS"
+)
+
+func AuthFlags(f *pflag.FlagSet) {
+	authOICDCAFile(f)
+	authOICDIssuerServer(f)
+	AuthOICDClaimUsername(f)
+	AuthOICDClaimGroups(f)
+}
+
+// Use only on generate-token utils 
+func AuthOICDCertKeyFile(f *pflag.FlagSet) {
+	desc := fmt.Sprintf(`OpenID Connect CA Cert filepath.
+Environment variable: %q`, authOICDCAKeyFileEnv)
+	f.String(authOICDCAKeyFileFlag, authOICDCAKeyFileDefault, desc)
+	_ = viper.BindPFlag(AuthOICDCAKeyFileViperKey, f.Lookup(authOICDCAKeyFileFlag))
 }
+
+func authOICDCAFile(f *pflag.FlagSet) {
+	desc := fmt.Sprintf(`OpenID Connect CA Key filepath.
+Environment variable: %q`, authOICDClaimUsernameEnv)
+	f.String(authOICDClaimUsernameFlag, authOICDClaimUsernameDefault, desc)
+	_ = viper.BindPFlag(authOICDClaimUsernameViperKey, f.Lookup(authOICDClaimUsernameFlag))
+}
+
+func authOICDIssuerServer(f *pflag.FlagSet) {
+	desc := fmt.Sprintf(`OpenID Connect issuer server domain (ie. https://quorum-key-manager.eu.auth0.com).
+Environment variable: %q`, authOICDIssuerURLEnv)
+	f.String(authOICDIssuerURLFlag, authOICDIssuerURLDefault, desc)
+	_ = viper.BindPFlag(authOICDIssuerURLViperKey, f.Lookup(authOICDIssuerURLFlag))
+}
+
+func AuthOICDClaimUsername(f *pflag.FlagSet) {
+	desc := fmt.Sprintf(`Token path claims for username.
+Environment variable: %q`, authOICDClaimGroupEnv)
+	f.String(authOICDClaimGroupFlag, authOICDClaimGroupDefault, desc)
+	_ = viper.BindPFlag(authOICDClaimGroupViperKey, f.Lookup(authOICDClaimGroupFlag))
+}
+
+func AuthOICDClaimGroups(f *pflag.FlagSet) {
+	desc := fmt.Sprintf(`Token path claims for groups.
+Environment variable: %q`, authOICDCACertFileEnv)
+	f.String(authOICDCACertFileFlag, authOICDCACertFileDefault, desc)
+	_ = viper.BindPFlag(authOICDCACertFileViperKey, f.Lookup(authOICDCACertFileFlag))
+}
+
+func NewAuthConfig(vipr *viper.Viper) (*auth.Config, error) {
+	var oicdCfg = &oicd.Config{}
+	
+	certs := []string{}
+	
+	fileCert, err := fileCertificate(vipr)
+	if err != nil {
+		return nil, err
+	} else if fileCert != "" {
+		certs = append(certs, fileCert)		
+	}
+	
+	issuerCerts, err := issuerCertificates(vipr)
+	if err != nil {
+		return nil, err
+	} else if issuerCerts != nil {
+		certs = append(certs, issuerCerts...)		
+	}
+
+	oicdCfg = oicd.NewConfig(vipr.GetString(authOICDClaimUsernameViperKey), 
+			vipr.GetString(authOICDClaimGroupViperKey), certs...)
+	
+	return &auth.Config{OICD: oicdCfg}, nil
+}
+
+func fileCertificate(vipr *viper.Viper) (string, error) {
+	caFile := vipr.GetString(authOICDCACertFileViperKey)
+	_, err := os.Stat(caFile)
+	if err != nil {
+		if !os.IsNotExist(err) {
+			return "", fmt.Errorf("failed to read CA file. %s", err.Error())
```

Comment:
should we use errors from the `errors` package?

Reply:
this will be handle by the CLI which does not have any errors parsing it won't mind